### ISSUE TITLE
amazon linux 2023 nodes wont join managed node group

### ISSUE THREAD

**Issue by tl-alex-nicot** [terraform-aws-modules/terraform-aws-eks#2954]
I have bumped the community module to 20.5.0 and begun work on upgrading my managed node group nodes to amazon linux 2023, but they wont join the cluster. 

I am attempting to run the latest linux 2023 arm64 image (amazon-eks-node-al2023-arm64-standard-1.29-v20240227)

I have checked we are running v1.16.2 of the aws vpc cni 

```
Couldn't proceed with upgrade process as new nodes are not joining node group karpenter-controller-20240206114028229100000002
```

I found in the userdata 

```
2024-03-05 15:47:45,667 - __init__.py[WARNING]: Unhandled unknown content-type (application/node.eks.aws) userdata: 'b'---'...'
```

in my mng terraform I set the following 

```
 ami_type                   = "AL2023_ARM_64_STANDARD"
      platform                   = "al2023"
      enable_bootstrap_user_data = true
      bootstrap_extra_args       = "--dns-cluster-ip 169.254.20.10" #NodeLocal ip
      pre_bootstrap_user_data    = <<-EOT
      #!/bin/bash
      modprobe ip_vs
      EOT
```

the userdata generated in the launch template looks fine regarding the required nodeadm work

```
Content-Type: multipart/mixed; boundary="MIMEBOUNDARY"
MIME-Version: 1.0

--MIMEBOUNDARY
Content-Transfer-Encoding: 7bit
Content-Type: application/node.eks.aws
Mime-Version: 1.0

---
apiVersion: node.eks.aws/v1alpha1
kind: NodeConfig
spec:
  cluster:
    name: experimental-eu-west-1
    apiServerEndpoint: [redacted]
    certificateAuthority: [redacted]
    cidr: 192.168.0.0/16

--MIMEBOUNDARY--
```

**Comment by bryantbiggs** [comment]
theres a few issues we're tracking down with this - and mainly we need the AMI types in the upstream provider:

- Provider https://github.com/hashicorp/terraform-provider-aws/pull/36046
- SDK [aws/aws-sdk-go-v2@release-2024-02-29/service/eks/CHANGELOG.md#v1410-2024-02-29](https://github.com/aws/aws-sdk-go-v2/blob/release-2024-02-29/service/eks/CHANGELOG.md?rgh-link-date=2024-03-01T23%3A24%3A48Z#v1410-2024-02-29)

When I added support, I validated from the basis of the user data based on the upstream AMI repository. I should have tried deploying an AL2023 based AMI in a nodegroup, but I missed that - apologies for that. 

There are a few other issues related as well such as determining if `cidr` is required for AL2023, etc. Its still early so there may be a few rough edges, but I think towards the end of this week we should have a better picture on resolutions
